evo_prot_grad/common/sampler.py

```diff
@@ -160,8 +159,17 @@ def _compute_gradients(self, ohs: List[torch.Tensor], PoE: torch.Tensor) -> torc
             # This checks whether the gradient sequence length
             # is exactly two more than the input sequence length.
             if oh_grad.shape[1] == self.chains_oh.shape[1] + 2:
-                oh_grad = oh_grad[:,1:-1]
-            summed_grads += [ oh_grad @ expert.expert_to_canonical_order ]
+                oh_grad = oh_grad[:, 1:-1]
+
+            # some tokenizers add an <end> token to the protein
+            # sequence like ProtT5 and Ankh, check and remove here
+            # if necessary.
+            # This checks whether the gradient sequence length
+            # is exactly one more than the input sequence length.
+            elif oh_grad.shape[1] == self.chains_oh.shape[1] + 1:
+                oh_grad = oh_grad[:, :-1]
```

Comment:
I noticed that we can't guarantee that the last element of the gradient sequence is what should removed (why not the first element?)